src/main/java/net/snowflake/client/core/arrow/DecfloatToRealConverter.java

```
@@ -0,0 +1,136 @@
+    if (isNull(rowIndex)) {
+      return 0;
+    }
+    BigDecimal bigDecimal = toBigDecimal(rowIndex);
+    if (bigDecimal.scale() == 0) {
+      short shortVal = bigDecimal.shortValue();
+      if (shortVal == bigDecimal.longValue()) {
+        return shortVal;
+      } else {
+        throw new SFException(
+            ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr, "Short", bigDecimal.toPlainString());
+      }
+    } else {
+      throw new SFException(
+          ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr, "Short", bigDecimal.toPlainString());
+    }
+  }
+
+  @Override
+  public int toInt(int rowIndex) throws SFException {
+    if (isNull(rowIndex)) {
+      return 0;
+    }
+    BigDecimal bigDecimal = toBigDecimal(rowIndex);
+    if (bigDecimal.scale() == 0) {
+      int intVal = bigDecimal.intValue();
+      if (intVal == bigDecimal.longValue()) {
+        return intVal;
+      } else {
+        throw new SFException(
+            ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr, "Integer", bigDecimal.toPlainString());
+      }
+    } else {
+      throw new SFException(
+          ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr, "Integer", bigDecimal.toPlainString());
+    }
+  }
+
+  @Override
+  public long toLong(int rowIndex) throws SFException {
+    if (isNull(rowIndex)) {
+      return 0;
+    }
+    BigDecimal bigDecimal = toBigDecimal(rowIndex);
+    return bigDecimal.longValue();
+  }
+
+  @Override
+  public Object toObject(int index) throws SFException {
+    if (isNull(index)) {
+      return null;
+    }
+    return toBigDecimal(index);
+  }
+
+  @Override
+  public String toString(int index) throws SFException {
+    if (isNull(index)) {
+      return null;
+    }
+    return toBigDecimal(index).toPlainString();
+  }
+
+  @Override
+  public byte[] toBytes(int index) throws SFException {
+    if (isNull(index)) {
+      return null;
+    }
+    BigDecimal val = toBigDecimal(index);
+    throw new SFException(
```

Hmm, I'm not sure I get it. I would throw an exception regardless of whether it is null or not, if we can't represent it as bytes.